source/server/hot_restarting_child.h

```diff
@@ -14,7 +14,35 @@ class HotRestartingChild : HotRestartingBase {
   HotRestartingChild(int base_id, int restart_epoch, const std::string& socket_path,
                      mode_t socket_mode);
 
+  // A structure to record the set of registered UDP listeners keyed on their addresses,
+  // to support QUIC packet forwarding.
+  class UdpForwardingContext {
+  public:
+    using ForwardEntry = std::pair<const Network::Address::Instance*, Network::UdpListenerConfig*>;
```

Comment:
It's scary to store address pointer here. Do you know its life time? I'm less concerned about UdpListenerConfig lifetime.

Or is it needed given it's the listener address in UdpListenerConfig?

Reply:
Sorry, UdpListenerConfig knows nothing about the listening socket. Could you clone the socket?

Reply:
It's needed because the call later is `UdpListenerConfig::listenerWorkerRouter(addr).deliver(...)` - the `UdpListenerConfig` instance contains multiple addresses.